components/script/dom/htmlscriptelement.rs

```diff
@@ -168,16 +260,43 @@ impl ScriptOrigin {
 
     pub fn external(text: DOMString, url: ServoUrl, type_: ScriptType) -> ScriptOrigin {
         ScriptOrigin {
-            text: text,
+            code: SourceCode::Text(text),
             url: url,
             external: true,
             type_,
         }
     }
 
     pub fn text(&self) -> DOMString {
-        self.text.clone()
+        match &self.code {
+            SourceCode::Text(text) => text.clone(),
+            SourceCode::Compiled(_) => panic!("Cannot convert a compiled script to text."),
```

Comment:
@gterzian Shall we replace this with the stored script as well? Or is that not needed?

Reply:
Yes we can do it here as well.

Reply:
Updated